File: backend/src/main/java/com/rapidphoto/features/gallery/api/dto/PagedPhotosResponse.java
package com.rapidphoto.features.gallery.api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Paginated response for photo listings.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PagedPhotosResponse {

    private List<PhotoListItemDto> content;

    private int page;

    private int size;

    private long totalElements;

    private int totalPages;

    private boolean hasNext;

    private boolean hasPrevious;
}

File: mobile/lib/features/gallery/models/paged_photos_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'photo_list_item.dart';

part 'paged_photos_response.freezed.dart';
part 'paged_photos_response.g.dart';

/// Paginated response for photo listings
@freezed
abstract class PagedPhotosResponse with _$PagedPhotosResponse {
  

  const factory PagedPhotosResponse({
    @Default([]) List<PhotoListItem> content,
    @Default(0) int page,
    @Default(20) int size,
    @Default(0) int totalElements,
    @Default(0) int totalPages,
    @Default(false) bool hasNext,
    @Default(false) bool hasPrevious,
  }) = _PagedPhotosResponse;

  factory PagedPhotosResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedPhotosResponseFromJson(json);
}
